### PR TITLE
refactor DockerSystem logic to make it support pass runner easy

### PR DESCRIPTION
**Pull request by guotuofeng**
## Checklist before requesting a review
- [ ] Add unit tests for this change.
- [ ] Make sure all tests can pass.
- [ ] Update documents if necessary.
- [ ] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.

### CODE REVIEW

File: olive/systems/utils.py [hunk 1]
```diff
@@ -2,66 +2,21 @@
 # Copyright (c) Microsoft Corporation. All rights reserved.
 # Licensed under the MIT License.
 # --------------------------------------------------------------------------
-import argparse
```
Comment (trajepl) [inline olive/systems/utils.py:5]: Curious what `misc` means?

File: olive/systems/python_environment/pass_runner.py [hunk 2]
```diff
@@ -21,7 +16,6 @@ def get_args(raw_args):
 
     parser.add_argument("--model_json_path", type=str, required=True, help="Path to input model json file")
     parser.add_argument("--pass_json_path", type=str, required=True, help="Path to pass json file")
-    parser.add_argument("--point_json_path", type=str, default=None, help="Path to point json file")
```
Comment (trajepl) [inline olive/systems/python_environment/pass_runner.py:24]: I am not very familar with this part of logics. QQ: what is the point_json_path? Is that safe to remove it? If so, please also update in PR description.

File: olive/systems/docker/utils.py [hunk 3]
```diff
@@ -19,19 +19,11 @@
...
 
 
-def create_config_file(
-    tempdir, model_config: "ModelConfig", metrics: List["Metric"], container_root_path: Path, model_mounts: dict
-):
-    model_json = model_config.to_json(check_object=True)
-    for k, v in model_mounts.items():
-        model_json["config"][k] = v
-
+def create_config_file(tempdir, config: dict, container_root_path: Path, model_mounts: dict):
```
Comment (trajepl) [inline olive/systems/docker/utils.py:22]: Can we also rename the `tempdir` lik what you did in `docker.system.py`?

File: olive/systems/azureml/aml_system.py [hunk 4]
```diff
@@ -718,6 +702,20 @@ def _create_metric_component(
         # metric component
         return cmd(**args)
 
+    def copy_code(self, code_files: List, code_root: Path):
```
Comment (trajepl) [inline olive/systems/azureml/aml_system.py:705]: nit: maybe rename `code_root` to `targer_code_path` or something else?

File: olive/systems/utils.py (hunk 1 above)
Comment (guotuofeng) [reply olive/systems/utils.py:5]: the file contains multiple utils and I could not find a better name for the file so I choose to misc.py

File: olive/systems/python_environment/pass_runner.py (hunk 2 above)
Comment (guotuofeng) [reply olive/systems/python_environment/pass_runner.py:24]: Move the config update to consistent with that one in AzureML system

File: olive/systems/azureml/aml_system.py (hunk 4 above)
Comment (guotuofeng) [reply olive/systems/azureml/aml_system.py:705]: will do

File: olive/systems/docker/utils.py (hunk 3 above)
Comment (guotuofeng) [reply olive/systems/docker/utils.py:22]: will change to workdir

**Review by jambayk — APPROVED** [review]

**Pull request merged**